File: Simulator/Views/ClientEditor.vb
Public Class ClientEditor

    Public Property Client As Models.Client

    Sub New()

        ' Esta llamada es exigida por el diseñador.
        InitializeComponent()

        ' Agregue cualquier inicialización después de la llamada a InitializeComponent().

        ' Grids
        Me.DataGridViewAccounts.AutoGenerateColumns = False
        Me.DataGridViewCards.AutoGenerateColumns = False

        Me.Client = New Models.Client

    End Sub

    Sub New(client As Models.Client)
        Me.New()
        Me.Client = client
    End Sub

    Private Sub ClientEditor_Load(sender As Object, e As EventArgs) Handles Me.Load
        Me.TextBoxFullName.Text = Client.Name
        Me.TextBoxIdentification.Text = Client.Identification
        Me.DataGridViewAccounts.DataSource = Client.Accounts
        Me.DataGridViewCards.DataSource = Client.Cards
    End Sub

    Private Sub ButtonCancel_Click(sender As Object, e As EventArgs) Handles ButtonCancel.Click
        Me.Parent.Controls.Remove(Me)
        Me.Dispose()
    End Sub

    Private Sub ButtonAccept_Click(sender As Object, e As EventArgs) Handles ButtonAccept.Click
        'SAVE
        RaiseEvent Saved(Me.Client, EventArgs.Empty)
        Me.ButtonCancel.PerformClick()
    End Sub

    Public Event Saved As EventHandler

    Private Sub TabControlClient_SelectedIndexChanged(sender As Object, e As EventArgs) Handles TabControlClient.SelectedIndexChanged
        If Me.TabControlClient.SelectedTab Is Me.TabPageCards Then
            Me.CollumnCardAccount1.DataSource = (From item In Client.Accounts
                                                 Where item.Type = "Ahorro"
                                                 Select item).ToList
            Me.CollumnCardAccount1.DisplayMember = "Number"
            Me.CollumnCardAccount1.ValueMember = "Number"
            Me.CollumnCardAccount2.DataSource = (From item In Client.Accounts
                                                 Where item.Type = "Corriente"
                                                 Select item).ToList
            Me.CollumnCardAccount2.DisplayMember = "Number"
            Me.CollumnCardAccount2.ValueMember = "Number"
        End If
    End Sub

    Private Sub DataGridViewCards_DataError(sender As Object, e As DataGridViewDataErrorEventArgs) Handles DataGridViewCards.DataError

    End Sub
End Class

File: Simulator-CSharp/Views/ClientEditor.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace ATM.Views
{
    public partial class ClientEditor : UserControl
    {
       
        public Models.Client Client { get; set; }
        
        public ClientEditor()
        {
            // Esta llamada es exigida por el diseñador.
            InitializeComponent();

            // Agregue cualquier inicialización después de la llamada a InitializeComponent().

            // Grids
            this.DataGridViewAccounts.AutoGenerateColumns = false;
            this.DataGridViewCards.AutoGenerateColumns = false;

            this.Client = new Models.Client();

        }

        public ClientEditor(Models.Client client) : this()
        {
            this.Client = client;
        }

        // ERROR: Handles clauses are not supported in C#
        private void ClientEditor_Load(object sender, EventArgs e)
        {
            this.TextBoxFullName.Text = Client.Name;
            this.TextBoxIdentification.Text = Client.Identification;
            this.DataGridViewAccounts.DataSource = Client.Accounts;
            this.DataGridViewCards.DataSource = Client.Cards;
        }

        // ERROR: Handles clauses are not supported in C#
        private void ButtonCancel_Click(object sender, EventArgs e)
        {
            this.Parent.Controls.Remove(this);
            this.Dispose();
        }

        // ERROR: Handles clauses are not supported in C#
        private void ButtonAccept_Click(object sender, EventArgs e)
        {
            //SAVE
            if (Saved != null)
            {
                Saved(this.Client, EventArgs.Empty);
            }
            this.ButtonCancel.PerformClick();
        }

        public event EventHandler Saved;

        private void TabControlClient_SelectedIndexChanged(Object sender, EventArgs e )
        {
            if (object.ReferenceEquals(this.TabControlClient.SelectedTab, this.TabPageCards))
            {
                this.CollumnCardAccount1.DataSource = (from item in Client.Accounts where item.Type == "Ahorro" select item).ToList();
                this.CollumnCardAccount1.DisplayMember = "Number";
                this.CollumnCardAccount1.ValueMember = "Number";
                this.CollumnCardAccount2.DataSource = (from item in Client.Accounts where item.Type == "Corriente" select item).ToList();
                this.CollumnCardAccount2.DisplayMember = "Number";
                this.CollumnCardAccount2.ValueMember = "Number";
            }
        }

        // ERROR: Handles clauses are not supported in C#
        private void DataGridViewCards_DataError(object sender, DataGridViewDataErrorEventArgs e)
        {
        }

    }
}
